src/adapter/src/optimize/materialized_view.rs

```diff
@@ -236,6 +236,14 @@ impl Optimize<LocalMirPlan> for Optimizer {
             sink_description,
         )?;
 
+        if self.config.mode == OptimizeMode::Explain {
```

Comment:
This and all similar `optimize/*.rs` changes will compute and trace `UsedIndexes` for the `OPTIMIZED` and `PHYSICAL` plans and record the result in the new `OptimizerTrace` layer under `optimize/global`.